### ISSUE TITLE
LoginPage - add props to make username and password optional

### ISSUE THREAD

**Issue by adamviktora** [patternfly/patternfly-react#11123]
**Is this a new component or an extension of an existing one?**
LoginPage

**Describe the feature**
Username and password form fields shoud not have the `required` prop hardcoded, but instead it should be configurable with props (`required` should still be the default though).

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
![Image](https://github.com/user-attachments/assets/23950890-ce48-44eb-b09b-90ae26e73b7a)

Original request from Slack: https://patternfly.slack.com/archives/C4FM977N0/p1729092084147469

**Comment by Mash707** [comment]
@thatblindgeye @tlabaj I would like work on this issue. From what I understand is that we are proposing to add props like `isUsernameRequired` and `isPasswordRequired` and passing it into `isRequired` 
https://github.com/patternfly/patternfly-react/blob/4a6b8820a7715ee4e6e1c663847c377b0421259b/packages/react-core/src/components/LoginPage/LoginForm.tsx#L86-L88

https://github.com/patternfly/patternfly-react/blob/4a6b8820a7715ee4e6e1c663847c377b0421259b/packages/react-core/src/components/LoginPage/LoginForm.tsx#L109

https://github.com/patternfly/patternfly-react/blob/4a6b8820a7715ee4e6e1c663847c377b0421259b/packages/react-core/src/components/LoginPage/LoginForm.tsx#L121

**Comment by thatblindgeye** [comment]
@adamviktora do you recall the original ask? The Slack link no longer works due to the conversation being older than 90 days on Slack.

I guess I'm wondering whether building out a custom LoginForm (using the various Form components) would make more sense than making login fields optional for a LoginPage (possibly more so if both username and password aren't required; at that point having the LoginForm as it is seems like an unnecessary step in the "login page" process).

**Comment by adamviktora** [comment]
Yeah, that makes sense. 

If I remember correctly, the original request was to make the password input optional, because they have a case where the password sometimes might be empty. I _think_ it was Carla Martinez who was originally asking.

**Comment by Mash707** [comment]
@thatblindgeye yup makes sense, having username and password optional contradicts the purpose of having a login page. 
We need more information regarding the original issue in slack.

**Comment by thatblindgeye** [comment]
@Mash707 I spoked with the consumer that originally requested this, and we should only have a prop to make password optional; username should still be required. Would you like to still work on this?

**Comment by Mash707** [comment]
@thatblindgeye Sure I'll work on this.
